packages/salesforce-adapter/test/change_validators/package_version.test.ts

```
@@ -0,0 +1,192 @@
+  ExactVersion,
+} from '../../src/change_validators/package_version'
+import { ACCOUNT_SETTINGS_METADATA_TYPE } from '../../src/constants'
+import { mockTypes } from '../mock_elements'
+import { createCustomObjectType } from '../utils'
+
+describe('when deploying objects with installed packages', () => {
+  const sbqq = new ReferenceExpression(
+    new ElemID('salesforce'),
+    new InstanceElement('SBQQ', mockTypes.InstalledPackage, {
+      versionNumber: '248.1',
+      fullName: 'SBQQ',
+      _alias: 'SBQQ',
+    }),
+  )
+  const sbaa = new ReferenceExpression(
+    new ElemID('salesforce'),
+    new InstanceElement('SBQQ', mockTypes.InstalledPackage, {
+      versionNumber: '244.0',
+      fullName: 'sbaa',
+      _alias: 'sbaa',
+    }),
+  )
+  const fakePackage = { resValue: { value: {} } }
+  let requiresExactVersion: boolean
+  describe.each(Object.entries(TYPES_PACKAGE_VERSION_MATCHING_EXACT_VERSION))(
+    'when package version in the instance is higher than in the target environment',
+    (instanceType: string, exactVersion: ExactVersion) => {
+      let change: Change
+      beforeEach(() => {
+        requiresExactVersion = exactVersion.exactVersion
+        const objectTypeToValidate = createCustomObjectType(instanceType, {
+          elemID: new ElemID('salesforce', instanceType),
+        })
+        const instance = new InstanceElement('asd', objectTypeToValidate, {
+          packageVersions: [
+            {
+              majorNumber: 249,
+              minorNumber: 1,
+              namespace: sbqq,
+            },
+            {
+              majorNumber: 244,
+              minorNumber: 1,
+              namespace: sbaa,
+            },
+            {
+              majorNumber: 1,
+              minorNumber: 7,
+              namespace: fakePackage,
+            },
+          ],
+        })
+        change = toChange({ after: instance })
+      })
+      it('should catch instance', async () => {
```

"should return an error for all packages which require a version match" or something like that.